### PR TITLE
add description for catoptioncombo viewType column headers

### PR DESCRIPTION
**Pull request by deeonwuli**
### :pushpin: References

-   **Issue:** Closes https://app.clickup.com/t/8693fnxz5

### :art: Screenshots
Screenshot 2024-01-11 at 15 43 02

### :fire: Notes to the tester
Datastore config:
```
"columnsDescriptions": {
    "Combined": "Only data that cannot be disaggregated into separate health sectors",
    "Community": "Only data from the community health workers",
    "Private": "Only data from the private health facilities",
    "Public": {
      "code": "PUBLIC"
    }
  }
```
where `Combined`, `Community`, `Private`, `Public` are the column names and `{code: "PUBLIC"}` is referencing a constant code. 

`REACT_APP_DHIS2_BASE_URL=https://dev.eyeseetea.com/who-dev-238/
`

### CODE REVIEW

**Review by MiquelAdell — APPROVED** [review]
It works for me. I created a test with another constant in 
https://dev.eyeseetea.com/who-dev-238/dhis-web-datastore/index.html#/edit/d2-autogen-forms/MAL-WMR
@eperedo if you have a moment to code review that.

Thank you both

**Review by eperedo — CHANGES_REQUESTED** [review]
thanks @deeonwuli for the code, functionality looks great, but I think using the code of a cat. combination is only going to work for this particular viewType and not in grids. For example in this section: `Completeness of outpatient reporting 2022`
image

In this case **Public, Private and community based** are just columns we create from the dataelement's formName. 

Maybe as a more generic implementation we can use the column name as an identifier instead of the combination code. 

what do you think? @deeonwuli @MiquelAdell

File: src/domain/common/entities/DataForm.ts [hunk 1]
```diff
@@ -47,6 +49,7 @@ export interface SectionBase {
     sortRowsBy: string;
     titleVariant: titleVariant;
     styles: SectionStyle;
+    categoryDescriptions: CategoryDescription;
```
[the remark this answers is not in the document]
Comment (MiquelAdell) [reply src/domain/common/entities/DataForm.ts:52]: the change of label is ok by me

File: src/webapp/reports/autogenerated-forms/GridWithCatOptionCombos.tsx [hunk 2]
```diff
@@ -131,10 +136,16 @@ const GridWithCatOptionCombos: React.FC<GridWithCatOptionCombosProps> = props =>
 };
 
 const useStyles = makeStyles({
-    wrapper: { margin: 10 },
-    header: { fontSize: "1.4em", fontWeight: "bold" as const },
+    header: {
+        fontSize: "1.2em",
+        fontWeight: "bold" as const,
```
[the remark this answers is not in the document]
Comment (deeonwuli) [reply src/webapp/reports/autogenerated-forms/GridWithCatOptionCombos.tsx:141]: the cast isn't absolutely necessary, i'll take it out.

**Review by eperedo — APPROVED** [review]
Everything looks good, thanks @deeonwuli.

**Pull request merged**